### PR TITLE
Vignettes/vectorcontrol

### PR DESCRIPTION
**Pull request by kellymccain28**
Updates to vector control vignettes: 

- Divided bednets and IRS into two vignettes: bednets is called VectorControl.Rmd and IRs is VectorControl_IRS.Rmd
- added explanation of `netz` package and that distribution != usage, etc. into bednets vignette
- added/edited text
- removed ggplot2 dependency
- added plot showing population of different mosquito species over time

*Question: For `set_spraying()`, where do theta and gamma come from for each of these parameters (mortality, feeding success, deterrence)? I'm not sure how to explain this in the vignette. Unsure about the distinction between "Matrix of mortality parameters" and "Matrix of mortality parameters per timestep".

### CODE REVIEW

File: vignettes/VectorControl.Rmd [hunk 1]
```diff
@@ -15,117 +15,124 @@ knitr::opts_chunk$set(
...
 
 ```{r setup}
-suppressPackageStartupMessages(library(ggplot2))
 library(malariasimulation)
 library(malariaEquilibrium)
-library(reshape2)
 ```
 
-# Parameterisation
+Long-lasting insecticidal bednets are highly effective and relatively cheap interventions to prevent malaria. The effects of insecticidal bed net distribution campsigns can be simulated using malariasimulation. The model provides the user with the flexibility to specify parameters that describe the net distribution campaign (e.g. timing, coverage, and target population), the bed nets distributed (e.g. efficacy, longevity), and the relative densities of the mosquito species present in the target population. We will illustrate this through an example with two bednet distributions, once per year.
```
Comment (pwinskill) [inline vignettes/VectorControl.Rmd:22]: Typo campaigns

Comment (pwinskill) [inline vignettes/VectorControl.Rmd:22]: I'd drop the sentence on mosqutio species - there are a lot more things characteristing the vectors than just relative density, so probably not something to go into here

File: vignettes/VectorControl.Rmd [hunk 2]
```diff
@@ -15,117 +15,124 @@ knitr::opts_chunk$set(
...
+  abline(v = bednetstimesteps, col = 'black', lty = 2, lwd = 2.5)
+  grid(lty = 2, col = 'grey80', lwd = 1)
+  legend('bottomleft', box.lty = 0, legend = c('Bednets','Prevalence for bednet scenario','Prevalence for control scenario'),
+         col = c('black', cols[3], cols[5]), lty = c(2,1,1), lwd = 2.5)
+}
+```
+  
+### Parameterisation
+
+Use the `get_parameters()` function to generate the list of parameters for a seasonal profile, accepting the default values to run the simulation from an equilibrium starting point. The seasonality will drive mosquito dynamics in the absence of interventions.
```
Comment (pwinskill) [inline vignettes/VectorControl.Rmd:47]: Seasonality isn't the focus of this vignette so I think should be turned off by default

File: vignettes/VectorControl.Rmd [hunk 3]
```diff
@@ -15,117 +15,124 @@ knitr::opts_chunk$set(
...
-    severe_incidence_rendering_max_ages = 10*year
+    # Here, we are asking for the output to contain information on severe incidence in children 2-10 years old.
+    clinical_incidence_rendering_min_ages = 2*year,
+    clinical_incidence_rendering_max_ages = 10*year
   )
 )
 
-simparams <- set_equilibrium(simparams, starting_EIR)
+# Setting 3 species of mosquitoes, 25% An. arabiensis, 25% An. funestus, and 50% An. gambiae
+simparams <- set_species(
```
Comment (pwinskill) [inline vignettes/VectorControl.Rmd:70]: ...likewise, I don't think mosqutio species/behavious should be the focus here either

File: vignettes/VectorControl.Rmd [hunk 4]
```diff
@@ -15,117 +15,124 @@ knitr::opts_chunk$set(
...
-  ks_gamma = matrix(c(0.008, 0.008), nrow=2, ncol=1),
-  ms_theta = matrix(c(-1.232, -1.232), nrow=2, ncol=1),
-  ms_gamma = matrix(c(-0.009, -0.009), nrow=2, ncol=1)
-)
+### Plot prevalence
+```{r, fig.align='center', fig.height=5, fig.width=7}
+plot_prev()
+```
+  
+## Visualisation
```
Comment (pwinskill) [inline vignettes/VectorControl.Rmd:107]: I thinkt his section is very helpful - can we call it something other than visualisation:

Something like:
`Comparing input coverage and population bed net usage`

File: vignettes/VectorControl.Rmd [hunk 5]
```diff
@@ -15,117 +15,124 @@ knitr::opts_chunk$set(
...
+```
+  
+## Using the `netz` package to estimate coverage inputs needed to achieve target population usage
 
-output <- run_simulation(sim_length, sprayingparams)
+The [`netz` package](https://mrc-ide.github.io/netz/index.html) is a useful tool to help set up bednets in `malariasimulation`. `malariasimulation` takes as input the proportion of the population who were distributed a bednet at specific time points, but available data for a specific region or country do not always have this information. `netz` allows for simple conversions between bednet metrics like crop, access, distribution, and usage.  
+  
+See the vignettes from the `netz` package that explain these concepts more comprehensively:  
+  
+* [This vignette](https://mrc-ide.github.io/netz/articles/overview.html) explains how to estimate the distribution coverage from a specified bednet usage.
```
Comment (pwinskill) [inline vignettes/VectorControl.Rmd:136]: I don't think we want to include this first point and link in this vignette as a lot of that is unrelated to the malariasimulation implementation. 

The second point is enough 👍

File: vignettes/VectorControl.Rmd [hunk 6]
```diff
@@ -1,8 +1,8 @@
 ---
-title: "Vector Control"
+title: "Vector Control: Bednets"
```
Comment (pwinskill) [inline vignettes/VectorControl.Rmd:2]: Might be helpful to rename the file as `VectorControl_bednets` given that the other is `VectorControl_IRS`

File: vignettes/VectorControl.Rmd (hunk 1 above)
Comment (pwinskill) [inline vignettes/VectorControl.Rmd:22]: Think easier not to make any statement here about the relative cost

File: vignettes/VectorControl_IRS.Rmd [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+month <- 30
+sim_length <- 3 * year
+human_population <- 1000
+starting_EIR <- 50
+
+simparams <- get_parameters(
+  list(
+    human_population = human_population,
+    # seasonality parameters
+    model_seasonality = TRUE, # Let's try a bi-modal model
```
Comment (pwinskill) [inline vignettes/VectorControl_IRS.Rmd:60]: Unlike with bed nets, seasonality is a defining factor of the IRS campaign, so it should be left in here 👍

Comment (pwinskill) [reply vignettes/VectorControl_IRS.Rmd:60]: To make it easier further on to work with the timing relative to the peak, we should probably use seasonality parmeters for a simple seasonal site, not bi-modal as the comment suggests.

File: vignettes/VectorControl_IRS.Rmd [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+output <- run_simulation(sim_length, sprayingparams)
+```
+
+
+### Plot prevalence
+```{r,fig.align='center', fig.height=5, fig.width=7}
+plot_prev()
+```
+
+## Additional species
```
Comment (pwinskill) [inline vignettes/VectorControl_IRS.Rmd:106]: This next section is really nice - but I think should be moved to a seperate vignette. It could focus on how we specify differnt species and binomics and then include the example of how this can alter intervention impact.

What do you think @htopazian ?

**Review by pwinskill — CHANGES_REQUESTED** [review]
Thanks for these @kellymccain28!
A few comments and editing suggestions

File: vignettes/VectorControl.Rmd [hunk 9]
```diff
@@ -15,117 +15,124 @@ knitr::opts_chunk$set(
...
-  ms_gamma = matrix(c(-0.009, -0.009), nrow=2, ncol=1)
-)
+### Plot prevalence
+```{r, fig.align='center', fig.height=5, fig.width=7}
+plot_prev()
+```
+  
+## Visualisation 
+
+It is important to understand the difference between the input `coverages` argument of `malariasimulation::set_bednets()` and the resulting population bednet usage over time in the model. In the above example, bednets are distributed to a specified percentage of the population at each distribution round; however, the level of bednet usage may be different.
```
Comment (htopazian) [inline vignettes/VectorControl.Rmd:109]: Last sentence re-arrange. 
When we set `coverages` in the above example to 0.8, we are telling the model to distribute bed nets to 80% of the population at year one, and to 50% of the population at year 2. However, the level of average bed net usage is not necessarily equal to 80% or 50%. Between these time points, bed net use will slowly decline over time (in reality this decline stems from things like disuse, holes in nets, lost nets, etc.).

File: vignettes/VectorControl.Rmd [hunk 10]
```diff
@@ -15,117 +15,124 @@ knitr::opts_chunk$set(
...
+* The size and frequency of distributions specified in `set_bednets()`  
+* The assumed net retention half life (`gamman`)
+* Correlations in the recipients of nets between rounds  
+  
+The output from `malariasimulation::run_simulation()` has a variable `n_use_net` that shows the number of people using bednets at any given timestep. We can visualise the proportion of the population using bednets over time to understand how bednet usage changes. 
+
+```{r, fig.align='center', fig.height=5, fig.width=7}
+output$prop_use_net <- output$n_use_net / human_population
+
+plot(x = output$timestep, y = output$prop_use_net, type = 'l',
```
Comment (htopazian) [inline vignettes/VectorControl.Rmd:122]: NICE

**Review by htopazian — CHANGES_REQUESTED** [review]
Great job! Adding a few comments as well

File: vignettes/VectorControl_IRS.Rmd (hunk 8 above)
Comment (kellymccain28) [reply vignettes/VectorControl_IRS.Rmd:106]: Ok, so what I've decided is to use only the default species = All for the bed net and IRS vignettes. I added a small section about the possibility of modelling multiple species and referenced the new set species vignette (need to add link later), and have created this new vignette for set_species().

File: vignettes/VectorControl.Rmd [hunk 11]
```diff
@@ -15,117 +15,124 @@ knitr::opts_chunk$set(
...
-  retention = 5 * year,
-  dn0 = matrix(c(.533, .45), nrow=2, ncol=1),
-  rn = matrix(c(.56, .5), nrow=2, ncol=1),
-  rnm = matrix(c(.24, .24), nrow=2, ncol=1),
-  gamman = rep(2.64 * 365, 2)
+  simparams,
+  timesteps = bednetstimesteps,
+  coverages = c(.8, .5),  # The first round is distributed to 80% of the population and the second round to 50%.
+  retention = 5 * year, # Nets are kept on average 5 years
+  dn0 = matrix(c(.533, .45), nrow=2, ncol=3), # Matrix of death probabilities for each mosquito species over time
```
[the remark this answers is not in the document]
Comment (kellymccain28) [reply vignettes/VectorControl.Rmd:93]: I changed the values so that they were the same for each timestep as I was not sure what the values corresponded to. The DOI in the documentation pointed me to the same paper that the IRS model was based on, which I'm not sure is correct since it doesn't seem to talk about nets. 
"The structure for the bed net model is documented in the S.I. of 10.1038/s41467-018-07357-w"

File: vignettes/SetSpecies.Rmd [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+```
+
+Here we can see that now there are 3 species of mosquitoes at the proportions we specified in `set_species()`.
+```{r}
+simparams$species
+simparams$species_proportions
+```
+
+  
+### Plot prevalence
```
Comment (pwinskill) [inline vignettes/SetSpecies.Rmd:156]: I'm not 100% sure I can see the difference in the impact of IRS that you are trying to demonstrate (these could just be stochastic differences).
It might be more clear if you compare 2 "example" species:
One with the defaults
One with a single species that has a significantly lower `phi_indoors` set.
That might make the comparison more obvious and also show the user that the vectors profiles can be modified?

**Review by pwinskill — CHANGES_REQUESTED** [review]
Thanks @kellymccain28 - these will be so helpful!
A few more points & Qus in this review

File: vignettes/SetSpecies.Rmd [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+         col = c("black", cols[3], cols[5]), lty = c(2,1,1), lwd = 2.5)
+}
+```
+    
+As alluded to in the IRS and Bednet vignettes, it is possible to account for varying proportions of mosquito species in the setting you are modelling using the `set_species()` function. IRS and bednets could be expected to have different impacts depending on the proportion of each mosquito species because of variations in indoor resting, insecticide resistance, and proportion of bites on humans versus animals by species. If you have specified more than 1 species, then the arguments for `set_spraying()` and/or `set_bednets()` must be populated with values for each species at each timestep that the intervention is implemented. 
+
+There are preset parameters for *An. gambiae*, *An. arabiensis*, and *An. funestus* that can be set by the helper functions `gamb_params()`, `arab_params()`, and `fun_params()`, respectively. The default values for each species in these helper functions are from 10.1073/pnas.1820646116. The parameters are:  
+
+* `blood_meal_rates`: the blood meal rates for each species 
+* `foraging_time`: time spent taking blood meals
```
[the remark this answers is not in the document]
Comment (kellymccain28) [reply vignettes/SetSpecies.Rmd:48]: I got these definitions from here: https://mrc-ide.github.io/malariasimulation/reference/get_parameters.html. It could be incorrect there though

File: vignettes/VectorControl_Bednets.Rmd [hunk 14]
```diff
@@ -0,0 +1,129 @@
...
+     col = cols[3], lwd = 2.5, ylim = c(0,1),
+     xlab = "Timestep (days)", ylab = "Proportion of population using bed nets",
+     xaxs = "i", yaxs = "i", bty = "l")
+grid(lty = 2, col = "grey80", lwd = 1)
+axis(side = 1, lty = 1, col = "black", pos = 0); axis(side = 2, lty = 1, col = "black")
+```
+  
+## Using the `netz` package to estimate coverage inputs needed to achieve target population usage
+
+The [`netz` package](https://mrc-ide.github.io/netz/index.html) is a useful tool to help set up bed nets in `malariasimulation`. `malariasimulation` takes as input the proportion of the population who were distributed a bed net at specific time points, but available data for a specific region or country do not always have this information. `netz` has functionality to estimate modelled population usage over time in the simulation for a given set of input coverages. It can also help to fit the input coverages to usage data.
```
[the remark this answers is not in the document]
Comment (htopazian) [reply vignettes/VectorControl_Bednets.Rmd:124]: "at specific time points, but net distribution or use data are not always available for a specific region or country"
"for a given set of input distributions. It can also ... "

File: vignettes/VectorControl_Bednets.Rmd [hunk 15]
```diff
@@ -0,0 +1,129 @@
...
+  
+See the vignettes from the `netz` package that explain these concepts more comprehensively:
```
Comment (htopazian) [inline vignettes/VectorControl_Bednets.Rmd:126]: Is there more than one?

File: vignettes/VectorControl_IRS.Rmd [hunk 16]
```diff
@@ -0,0 +1,113 @@
...
+
+```{r, fig.align="center", fig.height=4, fig.width=6}
+peak <- peak_season_offset(simparams)
+
+sprayingtimesteps <- c(1, 2) * year + peak - 3 * month # A round of IRS is implemented in the 1st and second year 3 months prior to peak transmission.
+
+sprayingparams <- set_spraying(
+  simparams,
+  timesteps = sprayingtimesteps,
+  coverages = c(0.3, 0.8), # # The first round covers 30% of the population and the second covers 80%.
```
Comment (htopazian) [inline vignettes/VectorControl_IRS.Rmd:96]: inconsistent with text above which says two rounds at 80% coverage

**Review by htopazian — CHANGES_REQUESTED** [review]
Hi Kelly, looking really good over all! 
My main organizational comment is that there is a lot of IRS included in the SetSpecies vignette. I think this is okay since in one you are outputting number of mosquitos and in the other you are outputting PfPR. But make sure all your notes are consistent between the two. You can also take out some of the explanation under "IRS with different mosquito species" and refer back to the VectorControl_IRS vignette

File: vignettes/SetSpecies.Rmd [hunk 17]
```diff
@@ -0,0 +1,180 @@
...
+
+Here we can see that now there are 3 species of mosquitoes at the proportions we specified in `set_species()`.
+```{r}
+simparams$species
+simparams$species_proportions
+```
+
+  
+### Plot prevalence
+In the plot below, we can see that IRS is much more effective when the default mosquito species are used compared to the scenario where we added an example mosquito species with a lower propensity for biting indoors. In this case, IRS will not be as effective because a larger proportion of bites take place outside of the home.
```
[the remark this answers is not in the document]
Comment (kellymccain28) [reply vignettes/SetSpecies.Rmd:161]: I've changed it so that there are equal proportions of each species

File: vignettes/VectorControl_Bednets.Rmd
Comment (NoSchID) [inline vignettes/VectorControl_Bednets.Rmd:1]: I have two suggestions on the bednet vignette:
 
1. I think it might be better to give an example of bednet distribution that we actually frequently run and that will illustrate the decay over time well. I suggest setting the sim_length to 6 years and distributing the same coverage (e.g. 50% or 80%) every 3 years.

2. Please add a reference to where the parameter values for resistance/other insecticides can be found. This is in: Table S1.3 in Supplementary Appendix 2 in https://www.sciencedirect.com/science/article/pii/S2542519621002965
This table includes the parameter values for pyrethroid-only and pyrethroid-PBO nets at various resistance levels.

File: vignettes/VectorControl_Bednets.Rmd [hunk 18]
```diff
@@ -0,0 +1,129 @@
...
+```
+
+### Plot prevalence
+```{r, fig.align="center", fig.height=5, fig.width=7}
+plot_prev()
+```
+  
+## Comparing input coverage and population bed net usage 
+
+It is important to understand the difference between the input `coverages` argument of `set_bednets()` and the resulting population bed net usage over time in the model. When we set coverages in the above example to 0.8, we are telling the model to distribute bed nets to 50% of the population at year one and to 50% of the population at year 4. However, the level of average bed net usage is not necessarily equal to 50%. Between these time points, bed net use will slowly decline over time (in reality this decline stems from things like disuse, holes in nets, lost nets, etc.).
```
Comment (pwinskill) [inline vignettes/VectorControl_Bednets.Rmd:103]: 0.8 and 50% don't align here?

File: vignettes/VectorControl_IRS.Rmd [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+  )
+)
+
+simparams <- set_equilibrium(parameters = simparams, init_EIR = starting_EIR)
+
+# Running simulation with no IRS
+output_control <- run_simulation(timesteps = sim_length, parameters = simparams)
+```
+
+#### A note on mosquito species
```
Comment (pwinskill) [inline vignettes/VectorControl_IRS.Rmd:73]: Can this be identical to the section in LLINs pls?

File: vignettes/SetSpecies.Rmd [hunk 20]
```diff
@@ -0,0 +1,182 @@
...
+
+* `blood_meal_rates`: the blood meal rates for each species 
+* `foraging_time`: time spent taking blood meals    
+* `Q0`: proportion of blood meals taken on humans  
+* `phi_bednets`: proportion of bites taken in bed    
+* `phi_indoors`: proportion of bites taken indoors    
+
+We will demonstrate how to specify different mosquito species and how this could alter intervention impact using an example with IRS.
+
+## IRS with default mosquito settings
```
Comment (pwinskill) [inline vignettes/SetSpecies.Rmd:54]: I'm undecided as to the value of this additional example, I'd vote for doing one of the following:

1. Just removing this example and leaving the vignette only up to line 52, or
2. Altering the example so that it compares 2 runs, each with a single species, but with one bionomics parameter varying - eg. example species 1 has low indoor resting and example species 2 has high indoor resting. then I think you could simplify the outputs to a single plot showing the differnt impact of IRS in each example.

**Review by pwinskill — CHANGES_REQUESTED** [review]

File: vignettes/SetSpecies.Rmd (hunk 20 above)
Comment (kellymccain28) [reply vignettes/SetSpecies.Rmd:54]: Have chosen option 2 here

File: vignettes/VectorControl_Bednets.Rmd [hunk 21]
```diff
@@ -0,0 +1,129 @@
...
+```
+
+### Plot prevalence
+```{r, fig.align="center", fig.height=5, fig.width=7}
+plot_prev()
+```
+  
+## Comparing input coverage and population bed net usage 
+
+It is important to understand the difference between the input `coverages` argument of `set_bednets()` and the resulting population bed net usage over time in the model. When we set coverages in the above example to 0.5, we are telling the model to distribute bed nets to 50% of the population at year one and to 50% of the population at year 4. However, the level of average bed net usage is not necessarily equal to 50%. Between these time points, bed net use will slowly decline over time (in reality this decline stems from things like disuse, holes in nets, lost nets, etc.).
```
Comment (htopazian) [inline vignettes/VectorControl_Bednets.Rmd:103]: Indicate that we are telling the model to distribute bed nets to a _random_ 50% of the population at years 1 and 4. Then will hopefully make more sense why the average usage is above 50% in the plot below.

**Review by pwinskill — APPROVED** [review]
Super - I think that setSpecies example looks great now 🙏🦩🌈
